src/Controller/RestApiController.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Www\Controller;
+
+use Www\DataObject\RestResponseDo;
+use Www\Exception\RestException;
+use YapepBase\Config;
+use YapepBase\Controller\HttpController;
+use YapepBase\Exception\ControllerException;
+use YapepBase\Mime\MimeType;
+use YapepBase\Request\IRequest;
+use YapepBase\Response\IResponse;
+use YapepBase\View\RestTemplate;
+
+abstract class RestApiController extends HttpController
+{
+    /** @var RestResponseDo */
+    protected $restResponseDo;
+
+    /** @var string */
+    private $requestBody;
+
+    /** @var array|null */
+    private $requestData;
+
+    public function __construct(IRequest $request, IResponse $response)
+    {
+        parent::__construct($request, $response);
+        $this->restResponseDo = new RestResponseDo();
+    }
+
+    public function run($action)
+    {
+        try {
+            $this->requestBody = file_get_contents('php://input');
+
+            if ($this->requestBody) {
```

Comment:
I know this works properly, and kinda matter of taste. But I still don't like it. In a lot of cases it confuses the developer who has to check if the variable is supposed to store a boolean or just casting it.
Also as we are going for strict_types it's even less appropriate to use these implicit type castings, I'd prefer to use our old friend empty()